lib/fetch/request.js

```diff
@@ -220,7 +220,7 @@ class Request {
       const referrer = init.referrer
 
       // 2. If referrer is the empty string, then set request’s referrer to "no-referrer".
-      if (referrer === '') {
+      if (referrer === '' || referrer === 'no-referrer') {
```

Comment:
Bug?

I think the spec is here wrong or not clear.

If referrer is empty string, set request.referrer to no-referrer. But what if it is 'no-referrer', because we explicitly set it to no-referrer in makeRequest? Here it will be set to no-referrer, and then go to the else branch and try to parse the 'no-referrer' like an URL.
https://github.com/nodejs/undici/blob/3372aed902951f135d710d0fb430f39b8571e547/lib/fetch/request.js#L231

Obviously this will throw.

Reply:
I think it's more likely that the EventSource spec is wrong here, and rather than making the change in fetch, you should instead just set the referrer to an empty string, and let fetch handle it.